src/components/ContentHighlights/data/hooks.js

```diff
@@ -68,6 +100,18 @@ export function useContentHighlightsContext() {
     }));
   }, [setState]);
 
+  const deleteSelectedRowId = useCallback((rowId) => {
+    const x = getState.stepperModal.currentSelectedRowIds;
+    delete x[rowId];
```

Comment:
If we're need to rely on the previous state to set the updated state here, we should rely on the `prevState` argument in the callback function passed to `setState` per the React best practices. Since state setting is asynchronous, the `getState.stepperModal.currentSelectedRowIds` _could_ be different from the prev state coming through as the `s` argument in `setState`'s callback function below.

Perhaps consider something along the lines of:

```jsx
const deleteSelectedRowId = useCallback((rowId) => {
  setState((s) => {
    const updatedCurrentSelectedRowIds = s.stepperModal.currentSelectedRowIds;
    delete updatedCurrentSelectedRowIds[rowId];
    return {
      ...s,
      stepperModal: {
        ...s.stepperModal,
        currentSelectedRowIds: updatedCurrentSelectedRowIds,
      },
    };
  });
}, [setState]);
```

Reply:
After implementing the changes, the UI is showing the item is being removed but the test are failing now showing the content is not being removed. This is after implementing the changes ConfirmContentCard PR suggestions below. Putting the exact function above does not remove the items from the UI but removes them from the `selectedRowIds` updating the checkmark in step 2 but no re-rendering the page to show the removed items in the ConfirmContent Page. 

Might need to rubber duck on why this could be happening. It would also explain when I chose to return a null if the item `isDeleted` followed by a useEffect to match the deleted item to the aggregation key as a way to force a re-render after the item has been deleted.

Reply:
Ended up reverting back to what I originally had in both `ContentConfirmContentCard` and `hooks`. Changing hooks resulted in an open gap when deleting items, while changing ContentConfirmContentCard resulted in failing test showing that the content still remained in the UI. Changing both resulted in the content not being removed at all. 
:(

Reply:
Thanks for the rubber ducking session to resolve this issue.